app.go

```diff
@@ -149,7 +156,13 @@ func (a *App) Run() error {
 func (a *App) Stop() (err error) {
 	sctx := NewContext(a.ctx, a)
 	for _, fn := range a.opts.beforeStop {
-		err = fn(sctx)
+		if err != nil {
```

Comment:
what is the purpose of this code?

Reply:
if err happened;but next fn has no err;this code will cover history err;
````
for {
// first time ;err != nil 
err = fn(sctx) 

// next time err = nil 
err = nil 
}
// return nil;err had been covered.
return err 

```